db/migrate/20151011151130_create_course_user_achievements.rb

```diff
@@ -0,0 +1,13 @@
+class CreateCourseUserAchievements < ActiveRecord::Migration
+  def change
+    create_table :course_user_achievements do |t|
+      t.references :course_user
+      t.references :achievement, foreign_key: { references: :course_achievements }
+      t.datetime :obtained_at, null: false
```

Comment:
If there is a link between user and achievement, the achievement is automatically awarded. that's what is meant by join table right?